truffle/src/com.oracle.truffle.api/src/com/oracle/truffle/api/TruffleLanguage.java

```diff
@@ -1036,6 +1036,14 @@ protected Object getLanguageGlobal(C context) {
         return AccessAPI.engineAccess().createDefaultLexicalScope(node, frame);
     }
 
+    protected List<String> getCompletionTriggerCharacters() {
```

Comment:
These methods need serious documentation.

Reply:
From the last call: the completion trigger characters should go into the language registration annotation so we can get all of them statically for all languages without loading the classes